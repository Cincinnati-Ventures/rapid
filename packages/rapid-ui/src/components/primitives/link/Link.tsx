import React from 'react';
import { RapidStyles } from '../../../utils';
type LinkType = 'a';
interface LinkProps extends HTMLAnchorElement {
	styles?: string;
	as?: LinkType;
}

const THEME_CLASSNAME = 'rapid-link';

const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
	({ styles, as = 'a', ...rest }, ref) => {
		const Component = (type: LinkType) => {
			switch (type) {
				case 'a':
					return (
						<a
							ref={ref}
							className={RapidStyles(
								styles || rest.className,
								'', //defaultStyles
								THEME_CLASSNAME,
							)}
						/>
					);
			}
		};
		return Component(as);
	},
);

Link.displayName = 'Link';

export default Link;
