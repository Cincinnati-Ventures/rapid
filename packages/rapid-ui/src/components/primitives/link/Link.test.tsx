import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Link from './Link';

describe('Link', () => {
	it('should render correctly', () => {
		const { container } = render(<Link />);
		expect(container).toMatchSnapshot();
	});
});
describe('Link', () => {
	it('should render  with href correctly', () => {
		const { container } = render(
			<Link href='www.example.com'>click here</Link>,
		);
		expect(container).toMatchSnapshot();
	});
});
