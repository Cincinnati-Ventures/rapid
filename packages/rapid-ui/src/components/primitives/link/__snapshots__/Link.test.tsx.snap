// Vitest Snapshot v1

exports[`Link > should render  with href correctly 1`] = `
<div>
  <a
    class="rapid-link"
  />
</div>
`;

exports[`Link > should render correctly 1`] = `
<div>
  <a
    class="rapid-link"
  />
</div>
`;

exports[`Link > should render href correctly 1`] = `
<div>
  <a
    class="rapid-link"
  />
</div>
`;
